ios/StripeTerminalReactNative.swift

```diff
@@ -307,12 +321,17 @@ class StripeTerminalReactNative: RCTEventEmitter, DiscoveryDelegate, BluetoothRe
 
         let locationId = params["locationId"] as? String
         let onBehalfOf: String? = params["onBehalfOf"] as? String
-
-        let connectionConfig = LocalMobileConnectionConfiguration(
-            locationId: locationId ?? selectedReader.locationId ?? "",
-            merchantDisplayName: nil, // use the location name,
-            onBehalfOf: onBehalfOf ?? nil
-        )
+        
+        let connectionConfig: LocalMobileConnectionConfiguration
+        do {
+            connectionConfig = try LocalMobileConnectionConfigurationBuilder(locationId: locationId ?? selectedReader.locationId ?? "")
+                .setMerchantDisplayName(nil) // use the location name
```

Comment:
we should support users passing a merchant display name. there is also a [tosAcceptancePermitted](https://stripe.dev/stripe-terminal-ios/docs/Classes/SCPLocalMobileConnectionConfiguration.html#/c:objc(cs)SCPLocalMobileConnectionConfiguration(py)tosAcceptancePermitted) field we should add 
cc @bric-stripe